Port to jbuilder

tmcgilchrist opened the pull request:
WIP branch for porting ocaml-aws to use jbuilder.

 - [x] port aws_gen to jbuilder
 - [x] port aws (core) to jbuilder
 - [x] generate jbuilder file for generated libraries
 - [x] fix code generation bug for duplicate constructors
 - [x] fix code generation bug for RDS message
 - [x] document how to build and generate new bindings
 - [x] document how to release a new version.

[1] for doing releases to opam https://github.com/samoht/dune-release

tmcgilchrist committed:
Output jbuild file template.

tmcgilchrist committed:
Fix compile errors with generated code.

tmcgilchrist committed:
Generating and building with jbuilder.

Commit:
Fix compile and remove .envrc

Commit:
compile-libraries

Commit:
Fix duplicate constructors

Commit:
Handle generating empty structs correctly

Commit:
Don't append empty query string on empty/None input_shape

gtrak commented:
I think this is done besides docs with #2 , and I'm not sure I know enough about opam releases to write the final doc items. 

I think the most relevant thing to my day work would be to generate and test s3 bindings, so I'm going to go down that path unless something else is more pressing?

tmcgilchrist commented:
Generating and testing the S3 bindings would be great. Any tests that could be added into that project would also be appreciated. I can setup AWS access for them from my account.

tmcgilchrist committed:
Rename jbuilder to dune.

tmcgilchrist committed:
Regenerate and fixup the jbuild for AWS tests.

tmcgilchrist committed:
Update to dune 1.0.

Use dune in opam files and cleanup unused files.

tmcgilchrist committed:
Update OPAM files for 2.0 and use dune 1.3.

Move opam files so dune considers this all one dune-project.

tmcgilchrist committed:
Use dune-release for publishing.

tmcgilchrist committed:
parent bb70057d8aa2f2de5fe32487da8058b1f97ffd62
author Tim McGilchrist <timmcgil@gmail.com> 1538434858 -0500
committer Tim McGilchrist <timmcgil@gmail.com> 1539724429 +1100

Build aws, aws-lwt, aws-async with travis.

Add PINS to travis ci build configuration.
Add specific package names into travis build.

tmcgilchrist committed:
Remove META files.

tmcgilchrist committed:
Fixup code gen to be compatible with 4.04.

Remove dune subst.

tmcgilchrist committed:
Cleanup README TODO's.

tmcgilchrist committed:
Build aws-gen.

tmcgilchrist commented:
Opened against the main project repo.
